iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs

```diff
@@ -71,23 +81,23 @@ private async Task RefreshLoopAsync(DateTime refreshesOn, CancellationToken canc
             {
                 if (Logging.IsEnabled) Logging.Info(this, refreshesOn, $"Before {nameof(RefreshLoopAsync)}");
 
-                if (waitTime.Seconds > 0)
+                if (waitTime > TimeSpan.Zero)
                 {
-                    await Task.Delay(waitTime, cancellationToken).ConfigureAwait(false);
+                    try
```

Comment:
I don't think exception handling should be done within this function. Rather the exceptions should be thrown and the task result observed. That way, the method is simpler to read and doesn't have 2 types of checks (exception and `IsCancellationRequested` for the same cancellationToken).